presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -5,35 +5,549 @@ Tuning Presto
+
+
+``query.manager-executor-pool-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``5``
+ * **Description:** Size of thread pool used for garbage collecting after queries. Threads from this pool are used to free resources from canceled queries, enforcing memory limits, queries timeouts etc. Higher number of threads will allow to manage memory more efficiently, so it may be increased to avoid out of memory exceptions in some scenarios. On the other hand higher value here may increase CPU usage for garbage collecting and use additional constant memory even if there is nothing to do for all of the threads.
+
+
+``query.min-expire-age``
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (duration)
+ * **Default value:** ``15 m``
+ * **Description:** This property describes time after which the query metadata may be removed from server. If value is low, it's possible that client will not be able to receive information about query completion. The value describes minimum time that must pass to remove query (after it's considered completed) but if there is space available in history queue the query data will be kept longer. The size of history queue is defined by ``query.max-history`` property (``100`` by default).
+
+
+``query.max-concurrent-queries``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Integer`` (at least ``1``)
+ * **Default value:** ``1000``
+ * **Description:** **Deprecated** Describes how many queries be processed simultaneously in single cluster node. It shouldn't be used in new configuration, the ``query.queue-config-file`` can be used instead.
+
+
+.. _query-max-memory:
+
+``query.max-memory``
+^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``20 GB``
+ * **Description:** Serves as default value for ``query_max_memory`` session property. This property also describes strict limit of total memory allocated around the cluster that may be used to process single query. The query is dropped if the limit is reached unless session want to prevent that by setting session property ``resource_overcommit``. The session may also want to decrease system pressure, so it's possible to decrease query memory limit for session by setting ``query_max_memory`` to smaller value. Setting ``query_max_memory`` to higher value then ``query.max-memory`` will not have any effect. This property may be used to ensure that single query cannot use all resources in cluster. The value should be set to be higher than what typical expected query in system will need - that way system will be resistant to SQL bugs that would cause large unwanted computation. Also if rare queries will require more memory, then the ``resource_overcommit`` session property may be used to break the limit. It is important to set this value to higher then default when presto runs complex queries on large datasets.
```

Comment:
Serves as the default value for the ``query_max_memory`` session property. This property also describes the strict limit of total memory that may be used to process a single query. A query is dropped if the limit is reached unless the``resource_overcommit`` session property is set.  This property helps ensure that a single query cannot use all resources in a cluster. It should be set higher than what is expected to be needed for a typical query in the system. It is important to set this to higher than the default if Presto will be running complex queries on large datasets. It is possible to decrease the query memory limit for a session by setting ``query_max_memory`` to a smaller value. Setting ``query_max_memory`` to a greater value than ``query.max-memory`` will not have any effect. 

I took out the bit about setting the value higher to be resistant to sql bugs that would cause large unwanted computation because that seemed like a reason to set it lower.